src/patterns/organisms/modal/modal.hbs

```diff
@@ -0,0 +1,60 @@
+---
+name: Modal
+description: |
+  A fullscreen modal dialog UI component.
+notes: |
+  A JS example to invoke a modal dialog is provided below.
+
+  The modal JS provides the following options
+    - `origin` [DOM Element] element that triggered the modal
+    - `content` [DOM ELement] content to display in the modal
+    - `options` [Object] object of optional params
+        - `title` [String] title to display at the top of the modal
+        - `onCreate` [Function] function to fire after modal has been created
+        - `onDestroy` [Function] function to fire after modal has been closed
+        - `allowScroll` [Boolean] allow/restrict page scrolling when modal is open
+        - `closeText` [String] text to use for close button a11y.
+tips: |
+  - You can use the utility class `mzp-u-modal-content` to hide content (when JS is enabled) that is intended to be shown only once a modal is displayed.
+  - Any HTML content you wish can be displayed within a modal.
+---
+
+<button class="mzp-c-button" type="button">Open Modal</button>
+
+<div class="mzp-u-modal-content">
+  <figure>
+    <!-- media could be an image or video -->
+    <img src="{{@root.baseurl}}/static/img/modal/modal.png" alt="">
+    <figcaption>
+      A description with a maximum of 100 characters.
+      That usually means only one or two sentences. <a href="#">Learn more.</a>
+    </figcaption>
+  </figure>
+</div>
+
+<script src="{{@root.baseurl}}/assets/protocol/protocol/js/protocol-modal.js"></script>
+
+<script>
+(function() {
+    'use strict';
+
+    var content = document.querySelector('.mzp-u-modal-content');
+    var button = document.querySelector('.mzp-c-button');
+
+    button.addEventListener('click', function(e) {
+        e.preventDefault();
+
+        Mozilla.Modal.createModal(e.target, content, {
+          title: 'Example headline with 35 characters',
+          closeText: 'Close modal',
```

Comment:
We probably need some better documentation around this, but we can at least comment that this is a demo and ideally one shouldn't have translatable strings in JS (unless you have a system for translating JS files or aren't localizing anyway). The title could be a heading in the DOM, closeText could be a data attribute, etc.

Reply:
I think it may be best (for now at least) to try and shy away from too much hand-holding when it comes to l10n, as different sites all have different methods of doing things. It's not directly related to the component itself, so to avoid over-complicating the instructions here, I think a generic l10n section may be a better time to explain this type of thing?

Reply:
Yeah, I'm probably overly paranoid about people following this example when hopefully it's clear that it's JUST an example. We're working on general l10n guidelines and "no strings in JS" is a good rule.